scarlet/component.py

```diff
@@ -363,8 +363,8 @@ def update(self):
         perform updates on multiple components at once
         (for example separating a buldge and disk).
         """
-        for component in self.components:
-            component.update()
+        for node in self._tree:
+            node.update()
```

Comment:
Good catch, looks like for a hierarchical tree we would have been calling update multiple times.

Reply:
I don't think so. We've called only the components, so nothing got done multiple times. The problem was that a node that is not a leaf (i.e. a multi-component source) never had its `update` method called.

Reply:
Ah, good point. You're right, I was thinking that it called `update` for all of the components, but when a source with multiple components had its `update` method called then some of them would be called again. I didn't notice that only the leaves were called.